flutter-for-android.md

```diff
@@ -0,0 +1,1864 @@
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    setState(() {
+      _lastLifecyleState = state;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_lastLifecyleState == null)
+      return new Text('This widget has not observed any lifecycle changes.', textDirection: TextDirection.ltr);
+    return new Text('The most recent lifecycle state this widget observed was: $_lastLifecyleState.',
+        textDirection: TextDirection.ltr);
+  }
+}
+
+void main() {
+  runApp(new Center(child: new LifecycleWatcher()));
+}
+{% endprettify %}
+
+# Layouts
+
+## What is the equivalent of a LinearLayout
+
+In Android, a LinearLayout is used to lay your widgets out linearly -horizontally 
+or vertically. In Flutter, you can use the Row widget or Column widget to 
+achieve the same result.
+
+If you notice the two code samples are identical with the exception of the 
+"Row" and "Column" widget. The children are the same and this feature can be 
+exploited to develop rich layouts that can change overtime with the same 
+children.
+
+<!-- skip -->
+{% prettify dart %}
+new Row(
+  mainAxisAlignment: MainAxisAlignment.center,
+  children: <Widget>[
+    new Text(
+      'Row One',
+    ),
+    new Text(
+      'Row Two',
+    ),
+    new Text(
+      'Row Three',
+    ),
+    new Text(
+      'Row Four',
+    ),
+  ],
+)
+{% endprettify %}
+
+<!-- skip -->
+{% prettify dart %}
+new Column(
+  mainAxisAlignment: MainAxisAlignment.center,
+  children: <Widget>[
+    new Text(
+      'Column One',
+    ),
+    new Text(
+      'Column Two',
+    ),
+    new Text(
+      'Column Three',
+    ),
+    new Text(
+      'Column Four',
+    ),
+  ],
+)
+{% endprettify %}
+
+## What is the equivalent of a RelativeLayout
+
+A RelativeLayout is used to lay your widgets out relative to each other. In 
+Flutter there are a few ways to achieve the same result.
+
+You can achieve the result of a RelativeLayout by using a combination of 
+Column, Row, and Stack widgets. You can specify rules for the widgets 
+constructors on how the children are laid out relative to the parent.
+
+[Simple example]
+
+You can also use the more complex but powerful CustomMultiChildLayout to 
+achieve a RelativeLayout.
+
+[complex code]
+
+## What is the equivalent of a ScrollView
+
+In Android a ScrollView lets you lay your widgets such that if the users' device has a 
+smaller screen than your content, they can scroll.
+
+In Flutter the easiest way to do this is using the ListView widget. This might 
+seem like overkill coming from Android, but in Flutter a ListView widget is 
+both a ScrollView and an Android ListView.
+
+<!-- skip -->
+{% prettify dart %}
+new ListView(
+  children: <Widget>[
+    new Text(
+      'Row One',
```

Comment:
you can remove the trailing commas on these strings inside the Text constructors. One line per new Text()